packages/database-persistence-layer/src/index.js

```diff
@@ -0,0 +1,2 @@
+export { default as create } from './create';
+export { default as __unstableConvertLegacyLocalStorageData } from './migrations/legacy-local-storage-data';
```

Comment:
I think the migration code has always been in `@wordpress/data` before this with the persistence plugin, but I don't think that was good because that package isn't concerned with the concrete structure of data. I think this is at least a step in the right direction, because this package contains lots of things specific to WordPress. There are also a few small changes to the data to make it work with the new persistence API.

I don't think `@wordpress/editor` is an option because it's unused by the widgets editors.

I did consider that each package that implements a preference could have migration code for that preference. That seems like the right ownership, but gluing all that together would be complicated.

I guess it could be an API like this:
```js
wp.preferences.registerMigration( oldData => newData );
```

Reply:
Is keeping it in `@wordpress/preferences` an option?

Reply:
`@wordpress/preferences` is not supposed to have concrete implementation code either, just like `@wordpress/data`. I think this new package is still a better option.

And I think it did need to be moved. The migration code was in the old persistence plugin, but that will become deprecated and unused, we wouldn't want to keep it active just for the purpose of doing a migration.